Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridDataConnection.cs

```diff
@@ -981,5 +1002,29 @@ private void UpdateDataProperties()
                 _dataProperties = null;
             }
         }
+
+        private void UpdateIncrementalItemsSource()
+        {
+            // Determine if incremental loading should be used
+            if (_dataSource is ISupportIncrementalLoading incrementalDataSource)
+            {
+                _incrementalItemsSource = incrementalDataSource;
+            }
+            else if (_owner.ItemsSource is ISupportIncrementalLoading incrementalItemsSource)
+            {
+                _incrementalItemsSource = incrementalItemsSource;
+            }
+            else
+            {
+                _incrementalItemsSource = default(ISupportIncrementalLoading);
+            }
+
+            _loadingOperation?.Cancel();
+        }
+
+        public IAsyncOperation<LoadMoreItemsResult> LoadMoreItemsAsync(uint count)
```

Comment:
Should be something like (in alphebetical order):

```
public void LoadMoteItems(uint count)
{
    Debug.Assert(_loadingOperation != null, "Expected non-null _loadingOperation.");
    _loadingOperation = _incrementalItemsSource.LoadMoreItemsAsync(count); 
    _loadingOperation.Completed += ... add an event handler in which you reset _loadingOperation to null.
}

```

Reply:
Should be `Debug.Assert(_loadingOperation == null, "Expected _loadingOperation == null.")`, right?

Reply:
Yes indeed. Sorry - my bad.